modules/core/src/main/java/org/apache/ignite/IgniteCompute.java

```diff
@@ -371,6 +371,9 @@ public void affinityRun(@NotNull Collection<String> cacheNames, int partId, Igni
      * <p>
      * If task for given name has not been deployed yet, then {@code taskName} will be
      * used as task class name to auto-deploy the task (see {@link #localDeployTask(Class, ClassLoader)} method).
+     * <p>
+     * If there are more one class deployed with the same name this method will execute the lasts one deployed of them.
+     * This method has no guarantees if it invokes where classes with the same names deployed from different threads.
```

Comment:
I suggest to reformulate javadoc like this:

`If class with the same name was deployed more than once, the last deployed version is used.
If method is called when other threads are deploying other versions of class with the same name there are no guarantees which version of the class will be executed.`